Modify OSS Airbyte and Cloud Airbyte to use Catalogs (#12346)
airbytehq/airbyte: docs/gradle-dependency-update.md

# Introduction

> This document helps to avoid tedious steps for updating **dependency** versions from old (non-relevant) to proper (relevant).    
> Here describes following procedure about how to update dependency versions for **Gradle** build and how to share them with other **projects**.      

Reviewer: ```suggestion
This document describes how to update dependency versions for Airbyte's **Gradle** build and how to share them with other **projects**.      
```

Dependencies should be represented as dependency coordinates, that a user can pick from when declaring dependencies in a build script.

> Version Catalog Example:
> ```gradle
> dependencies {
>    implementation(libs.groovy.core)
> }
> ```
> In this context, libs is a catalog and groovy represents a dependency available in this catalog. 

#### The version catalog TOML file format
Gradle offers a conventional file to declare a catalog.   

> It’s a conventional location where dependencies can be declared and then declaring in `settings.gradle` file.

Reviewer: ```suggestion
It’s a conventional location to declare dependencies that are both consumed and published.
```
